src/views/Graph.vue

```diff
@@ -333,6 +408,162 @@ export default {
+     *  PARENT_FAMILY: [ 'GRANDPARENT_FAMILY', 'GREAT_GRANDPARENT_FAMILY', 'root' ],
+     *  GRANDPARENT_FAMILY: [ 'GREAT_GRANDPARENT_FAMILY', 'root' ],
+     *  GREAT_GRANDPARENT_FAMILY : ['root'],
+     *  root: []
+     * }
+     *
+     * note: object value arrays do not contain family names as strings only - not nodes
+     *
+     * @returns {Object} keys are family names, values are arrays containing all ancestors names
+     */
+    allParentLookUp () {
+      const lookup = {}
+      this.namespaces.forEach((namespace) => {
+        const array = []
+        let parents = namespace.node.parents
+        while (parents.length) {
+          parents.forEach((parent) => {
+            const childTokens = this.workflows[0].tokens.clone({ cycle: `$namespace|${parent.name}` })
+            const childNode = this.cylcTree.$index[childTokens.id]
+            array.push(childNode.name)
+            parents = childNode.node.parents
+          })
+        }
+        lookup[namespace.name] = array
+      })
+      return lookup
+    },
+    /**
+     * Object for looking up children
+     *
+     * example return object
+     *
+     * {
+     *  workflow-name/run1/cycle/PARENT_FAMILY:
+     *   [
+     *     {
+     *       id: 'workflow-name/run1/cycle/PARENT_FAMILY'
+     *       name: 'PARENT_FAMILY'
+     *       children: Proxy
+     *     },
+     *     {
+     *       id: 'workflow-name/run1/cycle/FAMILY'
+     *       name: 'FAMILY'
+     *       children: Proxy
+     *     },
+     *     {
+     *       id: 'workflow-name/run1/cycle/task1'
+     *       name: 'task1'
+     *       children: Proxy
+     *     }
+     *   ]
+     *  workflow-name/run1/cycle/FAMILY:
+     *   [
+     *     {
+     *       id: 'workflow-name/run1/cycle/FAMILY'
+     *       name: 'FAMILY'
+     *       children: Proxy
+     *     },
+     *     {
+     *       id: 'workflow-name/run1/cycle/task1'
+     *       name: 'task1'
+     *       children: Proxy
+     *     }
+     *   ]
+     *  workflow-name/run1/cycle/task1:
+     *   [
+     *     {
+     *       id: 'workflow-name/run1/cycle/task1'
+     *       name: 'task1'
+     *       children: Proxy
+     *     }
+     *   ]
+     * }
+     * @returns {Object} keys are node ids, values are arrays of objects containing all children
+     */
+    allChildrenLookUp () {
+      const lookup = {}
+      // Function for getting a flattened array of the nested children
+      function childArray (a) {
+        return a.reduce(function (flattened, { id, name, children, type }) {
+          // We dont want to include jobs so dont include children if type is "task"
+          if (type === 'task') {
+            return flattened
+              .concat([{ id, name, type }])
+          } else {
+            return flattened
+              .concat([{ id, name, children, type }])
+              .concat(children ? childArray(children, id) : [])
+          }
+        }, [])
+      }
```

Comment:
This function can be extracted out of the computed property so it is not re-declared every time it runs. It would also allow unit testing of this function.

Reply:
Also it looks like `flatMap` would make this simpler:

```suggestion
      // Function for getting a flattened array of the nested children
      function childArray (a) {
        return a.flatMap(({ id, name, children, type }) => {
          if (type === 'task') {
            return [{ id, name, type }]
          } else {
            return [{ id, name, children, type }]
              .concat(children ? childArray(children) : [])
          }
        })
      }
```

Reply:
Done in 0908cd64b34900e51744daa7a7f080fd719846f4